service/state/index.go

```diff
@@ -144,18 +112,7 @@ func (i *Index) Events(height uint64, events []flow.Event) error {
 		}
 
 		for hash, evts := range buckets {
-			// Prefix + Block Height + Type Hash
-			key := make([]byte, 1+8+8)
-			key[0] = prefixDataEvents
-			binary.BigEndian.PutUint64(key[1:1+8], height)
-			binary.BigEndian.PutUint64(key[1+8:1+8+8], hash)
-
-			val, err := i.core.codec.Marshal(evts)
-			if err != nil {
-				return fmt.Errorf("could not encode events: %w", err)
-			}
-			val = i.core.compressor.EncodeAll(val, nil)
-			err = tx.Set(key, val)
+			err := SetEvents(height, hash, evts)(tx)
```

Comment:
`SaveEventsForHeight`?

Reply:
Here if we wanted to be consistent we should also mention the type, no? Like `SaveEventsForHeightAndType`, but it sounds very cumbersome.

Reply:
For now I changed it to `SaveEventsForHeight`.